src/storage/src/db_store.rs

```diff
@@ -181,23 +184,89 @@ impl DbStore {
         Ok(())
     }
 
+    //
+    // add document
+    //
+    fn add_document(
+        db: Pin<&mut Merk>,
+        sender: &DB3Address,
+        tx: &TxId,
+        mutation: &DatabaseMutation,
+        block_id: u64,
+        mutation_id: u32,
+    ) -> Result<()> {
+        let addr_ref: &[u8] = mutation.db_address.as_ref();
+        let db_id = DbId::try_from(addr_ref)?;
+        let database = Self::get_database(db.as_ref(), &db_id)?;
+        match database {
+            Some(d) => {
+                let mut entries: Vec<BatchEntry> = Vec::new();
+                let mut idx = 0;
+                let mut cid_index_map: HashMap<String, _> = HashMap::new();
+                for collection in d.collections.iter() {
+                    cid_index_map.insert(collection.name.to_string(), &collection.index_list);
+                }
+                for document_mutation in &mutation.document_mutations {
+                    if let Some(index_list) = cid_index_map.get(&document_mutation.collection_id) {
+                        for document in document_mutation.document.iter() {
+                            let document_id = DocumentId::create(block_id, mutation_id, idx)
+                                .map_err(|e| DB3Error::ApplyDatabaseError(format!("{:?}", e)))
+                                .unwrap();
+                            let db3_document =
+                                DB3Document::new(document.clone(), &document_id, &tx, &sender)
+                                    .map_err(|e| DB3Error::ApplyDatabaseError(format!("{:?}", e)))
+                                    .unwrap();
+                            let key_vec = document_id.as_ref().to_vec();
+                            let document_vec = db3_document.into_bytes().to_vec();
+                            idx += 1;
+                            entries.push((key_vec, Op::Put(document_vec)));
```

Comment:
here will store document multi-times ?

Reply:
multi-documents

Reply:
if the collection has multiple indexes, the db-store will store the document multi-times.
https://github.com/dbpunk-labs/db3/blob/888b2293a5de2e50579837e93aefd25bf16473c3/src/storage/src/db_store.rs#L210-L222

Reply:
i don't understand. 
```
message DocumentMutation {
    string collection_name = 1;
    repeated bytes document = 2;
}
```
for each document, we will store document only once. and store index multi-times if the collection define multi indexes